#2391 Map UUID to string

Blackbaud-JasonBodnar opened the pull request:
Fixes #2391

filiphr commented on processor/src/test/java/org/mapstruct/ap/test/conversion/uuid/UUIDConversionTest.java:
```diff
@@ -0,0 +1,70 @@
+@RunWith(AnnotationProcessorTestRunner.class)
+public class UUIDConversionTest {
+
+    private final GeneratedSource generatedSource = new GeneratedSource();
```
This is not used in this test. You can remove it entirely.

Blackbaud-JasonBodnar replied:
Fixed.

filiphr commented on processor/src/test/java/org/mapstruct/ap/test/conversion/uuid/UUIDConversionTest.java:
```diff
@@ -0,0 +1,70 @@
+        assertThat( source.getUUIDA() ).isEqualTo( UUID.fromString( target.getUUIDA() ) );
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```
Let's use `assertThatThrownBy` from AssertJ instead of `Test#expected`

Blackbaud-JasonBodnar replied:
Fixed.

filiphr reviewed (changes requested):
Thanks a lot for the PR @Blackbaud-JasonBodnar. 

It looks great. There are few small minor things that we should address before merging this. In addition to the comments I left can you also add a line to the documentation [here](https://github.com/mapstruct/mapstruct/blob/master/documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc#implicit-type-conversions). Similar to how it is done for `Currency` and `String`.

Blackbaud-JasonBodnar commented:
Sorry for the delay. We only get to work on outside projects once a month. I updated the docs and fixed the issues you mentioned. Hopefully, that's everything.

filiphr commented:
Thanks @Blackbaud-JasonBodnar I manually merged this in https://github.com/mapstruct/mapstruct/commit/0cb053df8d0d8cf3f1370f6afa76c98c5c62b3ce. 

We changed the way we test so had to adapt the test.

Thanks for the contribution
